<launch>
  <arg name="robot_name"        default="robot2"/>
  <!--map-->
  <arg name="localMap/width/float" value="14.0"/>
  <arg name="localMap/height/float" value="14.0"/>
  <arg name="localMap/resolution" value="0.05"/>
 <!-- rqt_reconfigure -->
 <arg name="groundEstimate/rqt_reconfigure" value="true"/>
 <arg name="classification/rqt_reconfigure" value="true"/>
 <arg name="imageMatching/rqt_reconfigure" value="true"/>
 <arg name="measurementVelocity/rqt_reconfigure" value="false"/>
 <arg name="estimationVelocity/rqt_reconfigure" value="true"/>
 <!-- output_screen-->
 <arg name="groundEstimate/output_screen" value="false"/>
 <arg name="classification/output_screen" value="false"/>
 <arg name="imageMatching/output_screen" value="false"/>
 <arg name="measurementVelocity/output_screen" value="false"/>
 <arg name="estimationVelocity/output_screen" value="true"/>
 
  <group ns="$(arg robot_name)">
    <!-- synchro and manage node -->
    <include file="$(find obstacle_detection_2019)/launch/synchroManage.launch"/>
    <!-- ground removement and clussification node -->
    <include file="$(find obstacle_detection_2019)/launch/remgroundClassification.launch">
      <!--map parameter-->
      <arg name="localMap/width/float" value="$(arg localMap/width/float)"/>
      <arg name="localMap/height/float" value="$(arg localMap/height/float)"/>
      <arg name="localMap/resolution" value="$(arg localMap/resolution)"/>
      <!-- rqt_reconfigure -->
      <arg name="groundEstimate/rqt_reconfigure" value="$(arg groundEstimate/rqt_reconfigure)"/>
      <arg name="classification/rqt_reconfigure" value="$(arg classification/rqt_reconfigure)"/>
      <!-- output_screen -->
      <arg name="groundEstimate/output_screen" value="$(arg groundEstimate/output_screen)"/>
      <arg name="classification/output_screen" value="$(arg classification/output_screen)"/>
    </include>
    <!-- image matching node -->
    <include file="$(find obstacle_detection_2019)/launch/imageMatching.launch">
      <arg name="imageMatching/debugType" value="1"/>
      <!--map parameter-->
      <arg name="localMap/width/float" value="$(arg localMap/width/float)"/>
      <arg name="localMap/height/float" value="$(arg localMap/height/float)"/>
      <arg name="localMap/resolution" value="$(arg localMap/resolution)"/>
      <!-- rqt_reconfigure -->
      <arg name="imageMatching/rqt_reconfigure" value="$(arg imageMatching/rqt_reconfigure)"/>
      <!-- output_screen-->
      <arg name="imageMatching/output_screen" value="$(arg imageMatching/output_screen)"/>
    </include>
    <!-- measurement velocity -->
    <include file="$(find obstacle_detection_2019)/launch/measurementVelocity.launch">
      <arg name="measurementVelocity/debugType" value="1"/>
      <!--debug parameter-->
      <arg name="measurementVelocity/timeRange" value="10"/>
      <arg name="measurementVelocity/timeInteval" value="1"/>
      <!-- rqt_reconfigure -->
      <arg name="measurementVelocity/rqt_reconfigure" value="$(arg measurementVelocity/rqt_reconfigure)"/>
      <!-- output_screen-->
      <arg name="measurementVelocity/output_screen" value="$(arg measurementVelocity/output_screen)"/>
    </include>
    <!-- estimation velocity -->
    <include file="$(find obstacle_detection_2019)/launch/velocityEstimation.launch">
      <arg name="estimationVelocity/debugType" value="1"/>
      <!--debug parameter-->
      <arg name="estimationVelocity/timeRange" value="10"/>
      <arg name="estimationVelocity/timeInteval" value="1"/>
      <!-- rqt_reconfigure -->
      <arg name="estimationVelocity/rqt_reconfigure" value="$(arg estimationVelocity/rqt_reconfigure)"/>
      <!-- output_screen-->
      <arg name="estimationVelocity/output_screen" value="$(arg estimationVelocity/output_screen)"/>
    </include>
  </group>

</launch>